adc/ad7779/ad7779-driver.c

```diff
@@ -436,13 +432,7 @@ static int dev_read(void *data, size_t size)
 	if (data != NULL && size != sizeof(unsigned))
 		return -EIO;
 
-	mutexLock(common.irq_lock);
-
-	condWait(common.irq_cond, common.irq_lock, 0);
-
-	*(uint32_t **)data = ((uint32_t *)&common.current);
-
-	mutexUnlock(common.irq_lock);
+	*(uint32_t **)data = (uint32_t *)&common.current;
```

Comment:
So we don't need synchronization with the interrupt via conditional?

Reply:
Well, with synchronization via conditional, the driver caused the thread reading it to be blocked
when no data was received or there were communication problems with AD7779. The common.current
is a read-only variable indicating the number of buffers read from the start. I have another concern here.
As it is passed by pointer (in order to send message only once and then simply checking the value), maybe
it would be more secure to pass it as value (but then again, the value will need to be checked by message
each time)?

Reply:
Added timeout on cond_wait